### PR TITLE
Add scraping for Cilium

### PR DESCRIPTION
**Pull request by whites11**
This PR adds a new scrape config to scrape cilium-agent and cilium-operator pods on management clusters running cilium app >= 0.4.0

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

**Comment by mcbenjemaa** [comment]
Please fix the build and the conflict

**Comment by whites11** [comment]
> Please fix the build and the conflict

done

File: go.mod [hunk 1]
```diff
@@ -126,7 +126,11 @@ replace (
 	// overriding this dependency to fix a security issue.
 	// see https://ossindex.sonatype.org/vulnerability/sonatype-2022-5436
 	github.com/labstack/echo/v4 => github.com/labstack/echo/v4 v4.9.0
+
+	github.com/nats-io/nats-server/v2 => github.com/nats-io/nats-server/v2 v2.9.3
```
Comment (mcbenjemaa) [inline go.mod:130]: Why changes in go.mod?

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml [hunk 2]
```diff
@@ -1206,3 +1206,31 @@
   - replacement: prometheus
     target_label: app
 [[ include "_labelingschema" . | indent 2 ]]
+[[- if eq .ClusterType "management_cluster" ]]
+# cilium agent and cilium operator
+- job_name: [[ .ClusterID ]]-prometheus/cilium-[[ .ClusterID ]]/0
```
Comment (mcbenjemaa) [inline files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:1211]: If this is tested?

File: go.mod (hunk 1 above)
Comment (whites11) [reply go.mod:130]: nancy

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml (hunk 2 above)
Comment (whites11) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:1211]: it's been running on gremlin, yeah

File: go.mod (hunk 1 above)
Comment (mcbenjemaa) [reply go.mod:130]: It’s already there!

Comment (mcbenjemaa) [reply go.mod:130]: Line 131

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml (hunk 2 above)
Comment (whites11) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:1211]: up

File: go.mod [hunk 3]
```diff
@@ -126,7 +126,11 @@ replace (
...
 	github.com/nats-io/nats-server/v2 v2.8.4 => github.com/nats-io/nats-server/v2 v2.9.3
```
Comment (QuentinBisson) [inline go.mod:131]:
```suggestion
```

File: go.mod [hunk 4]
```diff
@@ -126,7 +126,10 @@ replace (
 	// overriding this dependency to fix a security issue.
 	// see https://ossindex.sonatype.org/vulnerability/sonatype-2022-5436
 	github.com/labstack/echo/v4 => github.com/labstack/echo/v4 v4.9.0
-	github.com/nats-io/nats-server/v2 v2.8.4 => github.com/nats-io/nats-server/v2 v2.9.3
+
+	github.com/nats-io/nats-server/v2 => github.com/nats-io/nats-server/v2 v2.9.3
+
+	golang.org/x/text => golang.org/x/text v0.3.8
```
Comment (mcbenjemaa) [inline go.mod:132]: Please clean the old one 🙂

Comment (mcbenjemaa) [reply go.mod:132]: Below

Comment (QuentinBisson) [reply go.mod:132]: Done

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**